### ISSUE TITLE
Video resolutions in wrong order in player

### ISSUE THREAD

**Issue by SimplyCorbett** [Chocobozzz/PeerTube#3506]
On the latest git build sometimes video resolutions show up either in the wrong order or they show up in reverse. It's probably happening every one in thirty videos or something.

Example:

![firefox_ET2fExQBx6](https://user-images.githubusercontent.com/7935624/102878051-878e4680-43fc-11eb-9b70-4a5b825e0a10.png)

------
**Steps to reproduce:**

1. Import https://youtu.be/kdlZNAXxOVo
2. Video formats should be in the wrong order now. Have reproduced twice with this video.

**Describe the expected behavior**

Video formats in the correct order

**Additional information**

* PeerTube instance:
  * URL: https://vidrot.com
  * version: git
  * NodeJS version: 12
  * Ffmpeg version: 4.3.1

* Browser name, version and platforms on which you could reproduce the bug: Firefox 84

**Comment by Chocobozzz** [comment]
Hello,

Please fill the template

**Comment by SimplyCorbett** [comment]
Filled, updated.

**Comment by Chocobozzz** [comment]
Thanks, what are your transcoding options? In you screen, is it an HLS video?

**Comment by SimplyCorbett** [comment]
Yes, HLS is the only option enabled. Transcoding is set to 8 thread count with 480p, 720p and 1080p enabled.

Allow additional extensions & allow audio files upload are also enabled.

**Comment by SimplyCorbett** [comment]
Here is another video with the same issue - https://youtu.be/Z06RRp81iDM

https://i.imgur.com/MhDlSpK.png

I'm just randomly adding videos to see how peertube performs. It's not as frequent as I thought.

Hopefully you can find the bug but if not it's not a huge deal.

**Comment by Chocobozzz** [comment]
The input file has a very low bitrate. PeerTube bitrates per for 720p and 1080p are higher, so we just keep the original bitrate. Then sometimes 720p bitrate could be a little bit more important than 1080p (since we cap to the same bitrate). It's not a big issue, but we should try to fix it. Not sure how to handle these case however :thinking:

**Comment by SimplyCorbett** [comment]
It automatically picks 1080p anyway which was my main concern. Not sure why it's listing the order wrong if it's picking 1080p though.

**Comment by King-Noctis** [comment]
@Chocobozzz, I have an idea, let's say the video's original bitrate is 480p, could we list it in the usual order (while highlighting the original in parenthesis)?:
- 1080p
- 720p
- **480p (Original)**
- 360p
- **Auto**: should always prioritize the highest quality possible, and only downgrade to lower quality depending on internet bandwidth/speed. Though I'm not sure if transcoding a 480p video into 1080p will actually result in better quality?

**Comment by rigelk** [comment]
@King-Noctis that's why we cap resultions to the original video's.

**Comment by King-Noctis** [comment]
Sorry for my mistake @Chocobozzz and @rigelk, I mixed up video resolution and video bitrate. TBH, I’ve only heard about audio bitrate, I didn’t knew video also has bitrate. 

Video resolution and bitrate usually go hand-in-hand, higher resolution means higher bitrate, lower resolution means lower bitrate. But in rare cases of conflict, here's how you can choose to arrange them in rank of importance, “**Resolution > Bitrate > Framerate**”:  
- [ ] 4K:..............2160p; 14000kbps; 30fps  
- [ ] Full HD:....1080p; 8000kbps; 30fps 
- [ ] **HD:............720p; 4000kbps; 30fps** 
- [ ] **High:.........480p; _5000kbps_; 30fps** (–> although higher bitrate, it ranked lower because of lower resolution)
- [ ] High:.........480p; 1500kbps; 60fps
- [ ] **Medium:..360p; 1200kbps; 60fps** 
- [ ] **Low:..........270p; 400kbps; _240fps_** (–> although higher framerate, it ranked lower because of lower resolution)
- [ ] Low:..........270p; 400kbps; 120fps
- [ ] Auto

Why? I don’t know maybe because:
1. Mainstream and convention is more familiar with video resolution for video quality. 
2. Consistency for PeerTube. And also viewers from other video platforms like YouTube, Twitch, LBRY... 
3. Since PeerTube has more edutainment contents (maybe?), video’s details are more than playback smoothness?

Subjectively and from how I understand it, under limited internet bandwidth, I’d prioritize resolution for lectures or webinars, since clarity of text is more important. I’d prioritize bitrate for games and movies since action is more important than the details. I’d prioritize framerate for slow-motion scenes like sports.

Same goes for audio quality. Between a 1080p video with 192kbps audio versus a 720p video with 320kbps audio, the 1080p video should rank higher, because the majority usually prioritize visual over audio information. But of course this depends on content, sometimes like music or podcast, audio quality is more important.

**Comment by Zykino** [comment]
Just to add another example, I found the same bug on a video too : https://videos.pair2jeux.tube/videos/watch/c3446aa5-7a67-490b-aee1-d6ef1bee0551
![image](https://user-images.githubusercontent.com/3809938/109173503-309e4800-7784-11eb-96c2-b36a7e5f52c5.png)

I don’t think the video was uploaded from youtube, maybe @LecygneNoir could confirm.

From what I saw downloading Youtube’s videos from NewPipe, it happens that 60FPS videos files are lighters than 30FPS. Sometimes 1080p are lighter than 720p too. Not sure what you sort by nor if file size relate to bitrate in any way.
